From 564c2c35a7d176be5935051cd55149386f15254c Mon Sep 17 00:00:00 2001
From: grishche <a.grishchenko@intel.com>
Date: Tue, 6 Jun 2017 14:54:00 +0300
Subject: [PATCH 9730/9740] 
 mac80211-allow-to-get-the-vif-name-by-mac80211-drive

---
 include/net/mac80211.h | 13 +++++++++++++
 net/mac80211/util.c    | 16 ++++++++++++++++
 2 files changed, 29 insertions(+)

diff --git a/include/net/mac80211.h b/include/net/mac80211.h
index a3a2a72..ddde29e 100644
--- a/include/net/mac80211.h
+++ b/include/net/mac80211.h
@@ -1138,6 +1138,19 @@ struct ieee80211_vif *wdev_to_ieee80211_vif(struct wireless_dev *wdev);
 struct wireless_dev *ieee80211_vif_to_wdev(struct ieee80211_vif *vif);
 
 /**
+ * ieee80211_vif_to_name - return the vif name
+ * @vif: the vif to get the wdev for
+ *
+ * This can be used by mac80211 drivers with direct cfg80211 APIs
+ * (like the vendor commands) that needs to get the name for a vif.
+ *
+ * Note that this function may return %NULL if the given wdev isn't
+ * associated with a vif that the driver knows about (e.g. monitor
+ * or AP_VLAN interfaces.)
+ */
+const char *ieee80211_vif_to_name(struct ieee80211_vif *vif);
+
+/**
  * enum ieee80211_key_flags - key flags
  *
  * These flags are used for communication about keys between the driver
diff --git a/net/mac80211/util.c b/net/mac80211/util.c
index 5a4f0c4..2f76a03 100644
--- a/net/mac80211/util.c
+++ b/net/mac80211/util.c
@@ -649,6 +649,22 @@ struct wireless_dev *ieee80211_vif_to_wdev(struct ieee80211_vif *vif)
 }
 EXPORT_SYMBOL_GPL(ieee80211_vif_to_wdev);
 
+const char *ieee80211_vif_to_name(struct ieee80211_vif *vif)
+{
+	struct ieee80211_sub_if_data *sdata;
+
+	if (!vif)
+		return NULL;
+
+	sdata = vif_to_sdata(vif);
+
+	if (!sdata)
+		return NULL;
+
+	return &sdata->name;
+}
+EXPORT_SYMBOL_GPL(ieee80211_vif_to_name);
+
 /*
  * Nothing should have been stuffed into the workqueue during
  * the suspend->resume cycle. If this WARN is seen then there
-- 
2.10.1

